test/data/html/dom-text-scanner.html

```diff
@@ -17,11 +17,11 @@ <h1>DOMTextScanner Tests</h1>
         data-test-data='{
             "node": "div:nth-of-type(1)",
             "offset": 0,
-            "length": 15,
+            "length": 16,
             "expected": {
                 "node": "div:nth-of-type(2)>div::text",
                 "offset": 3,
-                "content": "小ぢん\nまり1\n小ぢん\nまり2"
+                "content": "\n小ぢん\nまり1\n小ぢん\nまり2"
```

Comment:
Okay the behavior of this one has changed for this one test because the startContainer is the first div, and so when it ends the `<div>小ぢん</div>`, it should append a `\n`. I feel like this makes sense.
```html 
<div><div>小ぢん</div>まり1</div>
<div>小ぢん<div>まり2</div></div>
```

The original `if (this._content.length > 0)` logic was trying to guard against this case, but it causes the bugs we've experienced in #1636 because a DOMTextScanner that starts between two elements would have omitted a `\n` that should've been there.